Assets/Scripts/UI/GameUI.cs

```diff
@@ -3,35 +3,101 @@
 using UnityEngine;
 using UnityEngine.UI;
 using Common;
+using UnityEngine.SceneManagement;
+using Inputs;
+
+public enum GameState { MainMenu, Game, GameOver }
 
 namespace UI
 {
     public class GameUI : Singleton<GameUI>
     {
-        [SerializeField] private Text playerScoreText, playerHealthText;
+        [SerializeField] private DefaultScriptableObject defaultScriptableObject;
+        [SerializeField] private Text playerScoreText, playerHealthText, hiScoreText, gameOverScoreText;
+        [SerializeField] private GameObject gameMenu, mainMenu, gameOverMenu;
+
+        private GameState gameState = GameState.MainMenu;
+
+        public GameState GameState { get { return gameState; }}
 
         private int playerScore = 0;
 
+        void OnEnable()
+        {
+            SceneManager.sceneLoaded += OnSceneLoaded;
+        }
+
+        void OnDisable()
+        {
+            SceneManager.sceneLoaded -= OnSceneLoaded;
+        }
+
+        private void OnSceneLoaded(Scene scene, LoadSceneMode mode)
+        {
+            switch (gameState)
```

Comment:
too much logic in the state change, this should be handled by their respective controller/view for each of the ui layers

Reply:
DOne